### ISSUE TITLE
Update JaCoCo version to allow Java 17

### ISSUE THREAD

**Issue by Thihup** [microsoft/azure-pipelines-tasks#15868]
## Required Information
*Type*: Bug
**Enter Task Name**: Maven

## Issue Description

The current version of JaCoCo injected in the pom.xml is version 0.8.4. However, it doesn't support using Java 17.
Currently, we're blocked to use Java 17 because of that. 
One quick fix would be to change the hardcoded version to 0.8.7 and a more flexible one would be allowing the user to specify the JaCoCo version, allowing the user to use the latest JaCoCo version to match the latest Java version.

JaCoCo version 0.8.4 is specified here: https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/codecoverage-tools/codecoverageconstants.ts#L213

**Comment by QiuyongWang** [comment]
Dear dev team, 
this is Frankie from MSFT DevOps support team. One of our premier customer is facing failure when trying to use azure devops yaml pipeline and mavenv3 task to create jacoco code coverage report. The error is known bug in jacoco-maven-plugin 0.8.4, and fixed in 0.8.7. So customer needs to use jacoco-maven-plugin 0.8.7 in azure pipeline. 

However, ADO pipeline agent always picks up the jacoco 0.8.4 to perform this mavenV3 task for coverage report.

After checking through this ADO mavenV3 task, it looks like this codecoverageconstants.ts has hard-coded the jacoco verion to 0.8.4. This matches customer pipeline error symptom - always the jacoco 0.8.4 is picked for the pipeline maven3 task for code coverage report.

Please correct me if any misunderstanding exists.

If above investigation is correct, appreciate you fix this issue as early as possible. The premier customer is blocked by this issue.

**Comment by KonstantinTyukalov** [comment]
Hi! Thanks for the information you provided. We are currently working on resolving this issue.

**Comment by KonstantinTyukalov** [comment]
Hi, the related changes will be available in the next release.
We will update the ticket once these changes will be fully rollout to all Azure DevOps users.

**Comment by QiuyongWang** [comment]
@KonstantinTyukalov @anatolybolshakov   Does it have a draft ETA plan to finish this bug fix deployment?

**Comment by KonstantinTyukalov** [comment]
@QiuyongWang The approximate waiting time for these changes to be available for all ADO users is 2-3 weeks.

**Comment by QiuyongWang** [comment]
@KonstantinTyukalov  Greetings. How's it going on this issue's fix deployment? Is there any place I could proactively to track that?

**Comment by KonstantinTyukalov** [comment]
The related changes were successfully rolled out for all Azure DevOps users, so closing this.
If you have any further questions, feel free to request reopening this issue.

**Comment by QiuyongWang** [comment]
@mhainke , wondering if you could go to another issue https://github.com/microsoft/azure-pipelines-tasks/issues/16200 , the developer is asking about some details on the multi-module maven project symptom on the maven task jacoco feature error.

**Comment by gianx001** [comment]
Hi, we have same problem but with Java 21 and jacoco 0.8.7, we have need upgrade to jacoco 0.8.11
